core/src/main/java/org/infinispan/manager/CacheManagerInfo.java

```diff
@@ -186,6 +193,37 @@ public boolean isTracingEnabled() {
                  .getGlobalConfiguration().tracing().enabled();
    }
 
+   public boolean allCachesHealthy() {
+      return cacheManager.getCaches().keySet().stream()
+            .allMatch(this::isCacheReady);
+   }
+
+   public boolean isCacheReady(String cacheName) {
+      try {
+         GlobalComponentRegistry gcr = SecurityActions.getGlobalComponentRegistry(cacheManager);
+         ComponentRegistry cr = gcr.getNamedComponentRegistry(cacheName);
+
+         log.debugf("Verifying %s, CR=%s", cacheName, cr);
+
+         // Component registry will be null if the cache is misconfigured or shutdown.
+         if (cr == null) return false;
+
+         // Verify if the component registry isn't in shutdown state.
+         if (!cr.getStatus().allowInvocations())
+            return false;
+
+         // Non-clustered caches accepting invocations will be ready.
+         if (!cr.getConfiguration().clustering().cacheMode().isClustered())
+            return true;
+
+         // Clustered caches are ready if they are not rebalacing.
+         DistributionManager dm = cr.getDistributionManager();
+         return dm != null && !dm.isRehashInProgress();
+      } catch (Exception ignore) {
+         return false;
```

Comment:
Should we log any exceptions here at the TRACE/DEBUG level? Do you know what exceptions could be possible here or is it just to be extra safe?

Reply:
Definitely don't bury the exception.

Reply:
I added the catch just to be safe, but it sounds good to log the exception.